src/options.js

```diff
@@ -29,21 +51,25 @@ function maybeTake(values, index) {
 }
 
 function maybeTypedMap(data, f, type) {
-  return map(data, type?.prototype instanceof TypedArray ? floater(f) : f, type);
+  return map(data, isNumberType(type) ? (d, i) => coerceNumber(f(d, i)) : f, type); // allow conversion from BigInt
```

Comment:
This still isn’t quite right.

The common case we were addressing with `floater` is when `f` returns a `BigInt`, and `type` is a common typed array type such as `Float64Array`. But this leaves a few other cases we aren’t handling correctly: `type` could be `BigInt64Array` or `BigUint64Array`, in which case we need to coerce using `BigInt` rather than `Number`; `type` could be an integer type such as `Int32Array`, in which case the `NaN` handling in `coerceNumber` is superfluous and unnecessary.

I’ll work on this now…

Reply:
I guess we don’t really need to support the case where `type` is `BigInt64Array` or `BigUint64Array`; or in other words we could require you to coerce to `BigInt` explicitly yourself. And the second case of `coerceNumber` doing superfluous coercion when `type` is `Int32Array` isn’t really a problem; it’s just slightly slower than it needs to be.